code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityExtension.java

```diff
@@ -13,232 +13,206 @@
 	}
 
 	/**
-	 * Calls {@link IdentityState#bootupIfReady(SharedStateCallback)} with a valid callback
-	 *
-	 * @return True if the bootup is complete
+	 * Responsible for handling the incoming events that this extension is registered for.
 	 */
-	boolean bootupIfReady() {
-		final SharedStateCallback callback = createSharedStateCallback();
-
-		return state.bootupIfReady(callback);
+	@VisibleForTesting
+	void handleEvent(@NonNull final Event event) {
+		if (EventUtils.isRequestIdentityEvent(event)) {
+			if (EventUtils.isGetUrlVariablesRequestEvent(event)) {
+				handleUrlVariablesRequest(event);
+			} else {
+				handleIdentityRequest(event);
+			}
+		} else if (EventUtils.isRequestContentEvent(event)) {
+			handleRequestContent(event);
+		} else if (EventUtils.isUpdateIdentityEvent(event)) {
+			handleUpdateIdentities(event);
+		} else if (EventUtils.isRemoveIdentityEvent(event)) {
+			handleRemoveIdentity(event);
+		} else if (EventUtils.isRequestResetEvent(event)) {
+			handleRequestReset(event);
+		} else if (EventUtils.isSharedStateUpdateFor(IdentityConstants.SharedState.IdentityDirect.NAME, event)) {
+			handleIdentityDirectECIDUpdate(event);
+		}
 	}
 
 	/**
 	 * Handles events requesting for formatted and encoded identifiers url for hybrid apps.
 	 *
 	 * @param event the identity request {@link Event}
 	 */
-	void handleUrlVariablesRequest(final Event event) {
-		String urlVariablesString = null;
-
-		final Map<String, Object> configurationState = getSharedState(
+	private void handleUrlVariablesRequest(final Event event) {
+		final SharedStateResult configSharedStateResult = sharedStateHandle.getSharedState(
 			IdentityConstants.SharedState.Configuration.NAME,
 			event
 		);
 
+		final Map<String, Object> configurationState = configSharedStateResult != null
+			? configSharedStateResult.getValue()
+			: null;
+
 		final String orgId = EventUtils.getOrgId(configurationState);
 
-		if (Utils.isNullOrEmpty(orgId)) {
+		if (StringUtils.isNullOrEmpty(orgId)) {
 			handleUrlVariableResponse(
 				event,
-				urlVariablesString,
+				null,
 				"IdentityExtension - Cannot process getUrlVariables request Identity event, Experience Cloud Org ID not found in configuration."
 			);
 			return;
 		}
 
-		ECID ecid = state.getIdentityProperties().getECID();
+		final ECID ecid = state.getIdentityProperties().getECID();
 		final String ecidString = ecid != null ? ecid.toString() : null;
 
-		if (Utils.isNullOrEmpty(ecidString)) {
+		if (StringUtils.isNullOrEmpty(ecidString)) {
 			handleUrlVariableResponse(
 				event,
-				urlVariablesString,
+				null,
 				"IdentityExtension - Cannot process getUrlVariables request Identity event, ECID not found."
 			);
 			return;
 		}
 
-		urlVariablesString =
-			URLUtils.generateURLVariablesPayload(String.valueOf(Utils.getUnixTimeInSeconds()), ecidString, orgId);
+		final String urlVariablesString = URLUtils.generateURLVariablesPayload(
+			String.valueOf(Utils.getUnixTimeInSeconds()),
```

Comment:
are there any core helpers for unixtimeinseconds?

Reply:
Good catch! Yes, `TimeUtils` is indeed publicly available for use and has this method. Will change it.